service/history/mutableStateBuilder.go

```diff
@@ -291,10 +300,86 @@ func (e *mutableStateBuilder) CloseUpdateSession() (*mutableStateSessionUpdates,
 		e.bufferedEvents = append(e.bufferedEvents, e.updateBufferedEvents)
 		e.updateBufferedEvents = nil
 	}
+	e.updateBufferedReplicationTasks = nil
+	e.deleteBufferedReplicationEvent = nil
 
 	return updates, nil
 }
 
+func (e *mutableStateBuilder) BufferReplicationTask(
+	request *h.ReplicateEventsRequest) *persistence.BufferedReplicationTask {
+	if _, ok := e.GetBufferedReplicationTask(request.GetFirstEventId()); ok {
```

Comment:
what if there are 2 active cluster?
2 can sent out a batch of events which have the same first event ID, i believe a comparison of failover version is needed?

Reply:
Yes.  It will be handled once I start implementing the merge protocol.